components/brave_federated_learning/brave_federated_learning_service.cc

```diff
@@ -0,0 +1,53 @@
+/* Copyright 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_federated_learning/brave_federated_learning_service.h"
+
+#include "bat/ads/pref_names.h"
+#include "brave/components/brave_federated_learning/brave_operational_profiling.h"
+#include "brave/components/brave_federated_learning/brave_operational_profiling_features.h"
+#include "brave/components/p3a/pref_names.h"
+#include "chrome/browser/profiles/profile_manager.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "components/prefs/pref_service.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "services/network/public/cpp/simple_url_loader.h"
+
+namespace brave {
+
+BraveFederatedLearningService::BraveFederatedLearningService(
+    PrefService* pref_service,
+    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory)
+    : pref_service_(pref_service), url_loader_factory_(url_loader_factory) {}
+
+BraveFederatedLearningService::~BraveFederatedLearningService() {}
+
+void BraveFederatedLearningService::RegisterLocalStatePrefs(
+    PrefRegistrySimple* registry) {
+  BraveOperationalProfiling::RegisterLocalStatePrefs(registry);
+}
+
+void BraveFederatedLearningService::Start() {
+  if (isAdsEnabled() && isP3AEnabled() && isOperationalProfilingEnabled()) {
```

Comment:
Are we not planning on restricting collection to US and non-Linux users?

Reply:
Correct. But since this feature is behind Griffin, our variation service, I thought that rather than having hardcoded restrictions we can directly restrict country and platform via Griffin, which is publicly verifiable ([here](https://griffin.brave.com/#)). Wdyt?

Reply:
Ah, makes sense